src/github-graphql-base.js

```diff
@@ -0,0 +1,600 @@
+import moment from 'moment';
+import GraphQL from 'graphql-client';
+
+import {
+  GITHUB_ISSUE_INFO_QUERY,
+  GITHUB_PR_INFO_QUERY,
+  GITHUB_LABEL_INFO_QUERY,
+  GITHUB_REACTION_INFO_QUERY
+} from '../script/queries/export';
+
+const DEBUG = process.env.NODE_ENV === 'development';
+
+function getTime(timeString) {
+  // parse timezone-aware time in string format to number
+  // e.g. '2017-01-01T00:00:00Z' becomes 1483228800000
+  // null means the earliest date
+  if (timeString) {
+    return moment(timeString).toDate().getTime();
+  } else {
+    return 0;
+  }
+}
+
+function sleep(s) {
+  return new Promise(resolve => setTimeout(resolve, s * 1000));
+}
+
+// A wrapper for GraphQL client that supports chainable methods.
+// Concurrency is NOT supported.
+// sample usages:
+// 1. fetchAllIssues = await Client.getGraphQLClient()
+//      .repo(repoOwner, repoName)
+//      .issues({sort: 'UPDATED_AT'}).
+//      .fetchAll({per_page: 100});
+// 2. fetchAllReactions = await Client.getGraphQLClient()
+//      .repo(repoOwner, repoName)
+//      .reactions({pr_number, per_review: 20})
+//      .fetchOne({per_page: 100});
+
+class GraphQLClientBase {
+  constructor(token, sleepTime = 3, warningThreshold = 15) {
+    this.token = token;
+    this.GH_GQL_BASE = 'https://api.github.com/graphql';
+    this.GH_GQL_OPTIONS = {
+      url: this.GH_GQL_BASE,
+      headers: token
+        ? { Authorization: `bearer ${token}` }
+        : {}
+    };
+    this.client = GraphQL(this.GH_GQL_OPTIONS);
+
+    // sleep time (sec) before timeout retry
+    this.sleepTime = sleepTime;
+
+    // maximum number of warnings before stop fetching
+    this.warningThreshold = warningThreshold;
+  }
+
+  repo(repoOwner, repoName) {
+    this.repoOwner = repoOwner;
+    this.repoName = repoName;
+    if (DEBUG) {
+      console.log('repoOwner set:', repoOwner, 'repoName set:', repoName);
+    }
+    return this;
+  }
+
+  // First type of queries: ISSUES
+  issues(config) {
+    const { sort, direction, earliestDate } = config || {};
+    // There are three types of order
+    // https://developer.github.com/v4/enum/issueorderfield/
+    // COMMENTS, CREATED_AT (default), and UPDATED_AT
+    this.orderBy = {
+      field: sort || 'CREATED_AT',
+      direction: direction || 'ASC'
+    };
+    this._fetch = this._fetchIssues;
+    this.earliestDate = earliestDate;
+    return this;
+  }
+
+  // Second type of queries: PULLREQUESTS
+  pullRequests(config) {
+    const { sort, direction, earliestDate,
```

Comment:
IMO the `sort ..` should go on a new line below, so that it lines up with the other items.